data_discovery_ai/utils/preprocessor.py

```diff
@@ -20,15 +20,43 @@
 from iterstrat.ml_stratifiers import MultilabelStratifiedShuffleSplit
 from tqdm import tqdm
 from pathlib import Path
+from typing import Dict
 
 # Base directory where your Poetry project's pyproject.toml is located
 BASE_DIR = Path(__file__).resolve().parent.parent
-SUB_DIR = BASE_DIR / "output"
+SUB_DIR = BASE_DIR / "temp"
```

Comment:
aren't the temp folder and files supposed to be on the fly (means, files are created/added to the system's temp directory) ? by saying "temp" shouldn't need to have pre-created `temp` folder and pre-prepared files inside. it is not dynamic yet.

Reply:
https://superuser.com/questions/332610/where-is-the-temporary-directory-in-linux

Reply:
so if you run the pipeline in your machine, it will generate the "temp" files dynamically to your system's temp directory, if I run on mine, likewise. 

the code will always try to generate the latest "temp" files during runtime if not available (you might have some sort of checks to avoid unnecessary attempts regenerating the temp files if nothing has really changed.)

at the end, someone else run your pipeline in different evironemtn dont' need to ask you for sending them "updated" "temp files" to download, if the code/parameters are modified and need to have new compatible "temp" files to work with.

Reply:
and another reasons to use system's temp directory is, when deploying your pipeline to different environment, the hard coded directory might cause permission errors etc. temp directory of operation system are always able to put new files in.